### PR TITLE
Sweets/cre8orscollective

### PR DESCRIPTION
**Pull request by SweetmanTech**
# Cre8ors Collective Passes
- Founders Pass (1-88)
- Collective Pass (tokenId: 89 - 888)
- `Cre8orsCollectiveMetadataRenderer` renders different metadata for tokens `Founders <= 88 < Collective`

### Mint Mechanism
- Founders Pass - Whitelist Mint - `purchasePresale`
- Collective Pass - Burn721Minter - `burn721Minter.purchase`

Screenshot 2023-05-02 at 6 56 29 PM

### CODE REVIEW

File: test/Cre8orsCollective.t.sol [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+    }
+
+    function getUri(uint256 tokenId) public view returns (string memory) {
+        string memory base = tokenId > 88
+            ? METADATA_BASE_COLLECTIVE
+            : METADATA_BASE_FOUNDERS;
+        return string(abi.encodePacked(base, Strings.toString(tokenId)));
+    }
+
+    function test_Purchase888Cre8orsCollectivePasses()
```
Comment (SweetmanTech) [inline test/Cre8orsCollective.t.sol:422]: this is the main unit test to focus on @crazzywizard

File: test/merkle/MerkleDrop.t.sol [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+        vm.expectRevert(IERC721Drop.Presale_Inactive.selector);
+        zoraNFTBase.purchasePresale{value: item.mintPrice}(
+            1,
+            item.maxMint,
+            item.mintPrice,
+            item.proof
+        );
+    }
+
+    function test_MerklePurchase88Tokens() public {
```
Comment (SweetmanTech) [inline test/merkle/MerkleDrop.t.sol:248]: main unit test to focus on @crazzywizard

File: src/metadata/Cre8orsCollectiveMetadataRenderer.sol [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        address target,
+        bytes32 provenanceHash
+    ) external requireSenderAdmin(target) {
+        provenanceHashes[target] = provenanceHash;
+        emit ProvenanceHashUpdated(target, provenanceHash);
+    }
+
+    /// @notice Update metadata base URI and contract URI
+    /// @param baseUri new base URI
+    /// @param newContractUri new contract URI (can be an empty string)
```
Comment (crazzywizard) [inline src/metadata/Cre8orsCollectiveMetadataRenderer.sol:83]: doc is missing baseCollectiveUri

File: src/metadata/Cre8orsCollectiveMetadataRenderer.sol [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            newContractUri,
+            0
+        );
+    }
+
+    /// @notice Update metadata base URI, extension, contract URI and freezing detailsUpdate metadata base URI, extension, contract URI and freezing detailsUpdate metadata base URI, extension, contract URI and freezing detailsUpdate metadata base URI, extension, contract URI and freezing detailsUpdate metadata base URI, extension, contract URI and freezing detailsUpdate metadata base URI, extension, contract URI and freezing detailsUpdate metadata base URI, extension, contract URI and freezing detailsUpdate metadata base URI, extension, contract URI and freezing details
+    /// @param target target contract to update metadata for
+    /// @param metadataBase new base URI to update metadata with
+    /// @param metadataExtension new extension to append to base metadata URI
+    /// @param freezeAt time to freeze the contract metadata at (set to 0 to disable)
```
Comment (crazzywizard) [inline src/metadata/Cre8orsCollectiveMetadataRenderer.sol:104]: doc missing metadataBaseCollective

File: src/metadata/Cre8orsCollectiveMetadataRenderer.sol [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+            metadataExtension,
+            newContractURI,
+            freezeAt
+        );
+    }
+
+    /// @notice Internal metadata update function
+    /// @param metadataBase Base URI to update metadata for
+    /// @param metadataExtension Extension URI to update metadata for
+    /// @param freezeAt timestamp to freeze metadata (set to 0 to disable freezing)
```
Comment (crazzywizard) [inline src/metadata/Cre8orsCollectiveMetadataRenderer.sol:126]: see comment above ⬆️

**Pull request merged**